plugins/modules/aci_vzany_to_contract.py

```diff
@@ -0,0 +1,319 @@
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+'''
+
+
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+from ansible.module_utils.basic import AnsibleModule
+
+
+ACI_CLASS_MAPPING = dict(
+    provider={
+        'class': 'vzRsAnyToProv',
+        'rn': 'rsanyToProv-',
+        'target_attribute': 'tnVzBrCPName'
+    },
+    consumer={
+        'class': 'vzRsAnyToCons',
+        'rn': 'rsanyToCons-',
+        'target_attribute': 'tnVzBrCPName'
+    },
+    interface={
+        'class': 'vzRsAnyToConsIf',
+        'rn': 'rsanyToConsIf-',
+        'target_attribute': 'tnVzCPIfName'
+    }
+)
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        tenant=dict(type='str', aliases=['tenant_name']),
+        vrf=dict(type='str', aliases=['context', 'vrf_name']),
+        contract=dict(type='str', aliases=['contract_name']),
+        contract_type=dict(type='str', required=True, choices=[
+                           'provider', 'consumer', 'interface']),
+        state=dict(type='str', default='present',
+                   choices=['absent', 'present', 'query'])
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['contract_type', 'contract', 'vrf', 'tenant']],
+            ['state', 'present', ['contract_type', 'contract', 'vrf', 'tenant']],
+            ['state', 'query', ['contract_type']]
```

Comment:
this is not needed as you marked contract_type to required so it is always required.